examples/sequence_tagging/conll_reader.py

```diff
@@ -0,0 +1,266 @@
+        sent = inst.sentence
+        data.append([sent.word_ids, sent.char_id_seqs, inst.ner_ids])
+        inst = reader.getNext(normalize_digits)
+
+    reader.close()
+    print("Total number of data: %d" % counter)
+    return data
+
+
+def iterate_batch(data, batch_size, shuffle=False):
+    if shuffle:
+        np.random.shuffle(data)
+
+    for start_idx in range(0, len(data), batch_size):
+        excerpt = slice(start_idx, start_idx + batch_size)
+        batch = data[excerpt]
+
+        batch_length = max([len(batch[i][0]) for i in range(len(batch))])
+
+        wid_inputs = np.empty([len(batch), batch_length], dtype=np.int64)
+        cid_inputs = np.empty([len(batch), batch_length, MAX_CHAR_LENGTH],
+                              dtype=np.int64)
+        nid_inputs = np.empty([len(batch), batch_length], dtype=np.int64)
+        masks = np.zeros([len(batch), batch_length], dtype=np.float32)
+        lengths = np.empty(len(batch), dtype=np.int64)
+
+        for i, inst in enumerate(batch):
+            wids, cid_seqs, nids = inst
+
+            inst_size = len(wids)
+            lengths[i] = inst_size
+            # word ids
+            wid_inputs[i, :inst_size] = wids
+            wid_inputs[i, inst_size:] = PAD_WORD
+            for c, cids in enumerate(cid_seqs):
+                cid_inputs[i, c, :len(cids)] = cids
+                cid_inputs[i, c, len(cids):] = PAD_CHAR
+            cid_inputs[i, inst_size:, :] = PAD_CHAR
+            nid_inputs[i, :inst_size] = nids
+            nid_inputs[i, inst_size:] = PAD_NER
+            masks[i, :inst_size] = 1.0
+
+        yield wid_inputs, cid_inputs, nid_inputs, masks, lengths
+
+
+def load_glove(filename, emb_dim, normalize_digits=True):
+    r"""Loads embeddings in the glove text format in which each line is
+    '<word-string> <embedding-vector>'. Dimensions of the embedding vector
+    are separated with whitespace characters.
+    """
+    glove_dict = dict()
+    with open(filename) as fin:
+        for line in fin:
+            vec = line.strip().split()
+            if len(vec) == 0:
+                continue
+            word, vec = vec[0], vec[1:]
+            word = tx.utils.compat_as_text(word)
+            word = DIGIT_RE.sub("0", word) if normalize_digits else word
+            glove_dict[word] = np.array([float(v) for v in vec])
+            if len(vec) != emb_dim:
+                raise ValueError("Inconsistent word vector sizes: %d vs %d" %
+                                 (len(vec), emb_dim))
+    return glove_dict
+
+
+def construct_init_word_vecs(vocab, word_vecs, glove_dict):
+    for word, index in vocab.items():
+        if word in glove_dict:
+            embedding = glove_dict[word]
+        elif word.lower() in glove_dict:
+            embedding = glove_dict[word.lower()]
+        else:
+            embedding = None
+
+        if embedding is not None:
+            word_vecs[index] = embedding
+    return word_vecs
+
+
+class CoNLLReader:
```

Comment:
Can we modify this to use `tx.data.Dataset`? This would eliminate the need for separate `read_data` and `iterate_batch` methods.